File: Data Structure&Algorithm/meeting/laohu/demo1.py
import sys
def main():
    res=set()
    line = sys.stdin.readline().strip()
    n,k=list(map(int, line.split()))
    line = sys.stdin.readline().strip()
    arr=list(map(int, line.split()))
    left,right=0,n-1
    while left<right:
        if arr[left]+arr[right]==k:
            res.add((arr[left],arr[right]))
            left+=1
            right-=1
        if arr[left]+arr[right]<k:
            left+=1
        if arr[left]+arr[right]>k:
            right-=1
    # print(res)
    res=list(res)
    res.sort()
    for item in res:
        print(item[0],item[1])
main()

File: Data Structure&Algorithm/meeting/laohu/demo1.cpp
#include<iostream>
#include<vector>
using namespace std;
int main()
{
    int n,k;
    vector<int> arr;
    cin>>n>>k;
    for(int i=0;i<n;i++)
    {
        int temp;
        cin>>temp;
        arr.push_back(temp);
    }
    int left=0;
    int right=n-1;
    while(left<right)
    {
        if(arr[left]+arr[right]==k){
            cout<<arr[left]<<" "<<arr[right]<<endl;
            left+=1;
            right-=1;
        }
        if(arr[left]+arr[right]<k)
            left+=1;
        if(arr[left]+arr[right]>k)
            right-=1;
    }
}